### PR TITLE
Add scaling and concurrency settings to VEBA UI

### PR DESCRIPTION
**Pull request by lamw**
Closes: #448
Signed-off-by: William Lam <wlam@vmware.com>

## Summary

This change introduces the ability to configure `minScale`, `maxScale` and `containerConcurrency` settings for a given function deployment using the VEBA UI

Screen Shot 2021-07-07 at 8 59 03 AM

Using values of 1,1,1 will result in Knative Service that looks like the following:

```console
...
Spec:
  Template:
    Metadata:
      Annotations:
        autoscaling.knative.dev/maxScale:  1
        autoscaling.knative.dev/minScale:  1
      Creation Timestamp:                  <nil>
    Spec:
      Container Concurrency:  1
....
```

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by embano1** [comment]
Are we somewhere documenting the UI settings? Bc a user might not understand that 0 is default ie no limits. I also don't see this stated in the screenshot.

**Comment by lamw** [comment]
> Are we somewhere documenting the UI settings? Bc a user might not understand that 0 is default ie no limits. I also don't see this stated in the screenshot.

The UI default values that are populated in the UI is `0` and I took these as "defaults". Its good point about adding this to documentation. I'll take a stab and finding the right place in our docs

File: docs/kb/use-functions.md [hunk 1]
```diff
@@ -88,6 +89,47 @@ Step 4 - Test your functions
...
+
+* `Docker Image` - The Docker image that contains the function logic
+
+Step 5 - (optional) In the `Configuration` section of the wizard, users can control the concurrency and scale bounds for the function:
+
+* `Container concurrency:` - Specifies the maximum allowed in-flight (concurrent) requests per container of the Revision. Defaults to 0 which means concurrency to the application is not limited, and the system decides the target concurrency for the autoscaler.
+
+* `Minimum scale:` - Controls the minimum number of replicas that each Revision should have. Knative will attempt to never have less than this number of replicas at any one point in time.
+
+* `Maximum scale:` - Controls the maximum number of replicas that each revision should have. Knative will attempt to never have more than this number of replicas running, or in the process of being created, at any one point in time.
```
Comment (embano1) [inline docs/kb/use-functions.md:112]: It doesn't say what "0" means.

**Review by embano1 — APPROVED** [review]
LGTM

**Pull request merged**